cmd/update.go

```diff
@@ -2,55 +2,97 @@ package cmd
 
 import (
 	"bufio"
+	"errors"
 	"fmt"
 	"github.com/blang/semver"
 	"github.com/rhysd/go-github-selfupdate/selfupdate"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 	"os"
 )
 
-func selfUpdate() {
-	current := version
-	latest, found, err := selfupdate.DetectLatest("screwdriver-cd/sd-local")
+var (
+	githubSlug     = "screwdriver-cd/sd-local"
+	currentVersion = version
+	updateFlag     = false
+)
+
+func canUpdate() (*selfupdate.Release, error) {
+
+	if currentVersion == "dev" {
+		err := errors.New("This is a development command and cannot be updated.")
+		return &selfupdate.Release{}, err
+	}
+
+	latest, found, err := selfupdate.DetectLatest(githubSlug)
 	if err != nil {
-		fmt.Println("Error occurred while detecting version:", err)
-		return
+		return &selfupdate.Release{}, err
 	}
-	fmt.Println("Current version:", current)
-	if current == "dev" {
-		return
+	if !found {
+		err = errors.New("Repositry Not Found")
+		return &selfupdate.Release{}, err
 	}
-	v := semver.MustParse(current)
-	if !found || latest.Version.LTE(v) {
-		fmt.Println("Current version is the latest")
-		return
+	v := semver.MustParse(currentVersion)
+	if latest.Version.LTE(v) {
+		err = errors.New("Current version is latest")
+		return &selfupdate.Release{}, err
 	}
-	fmt.Print("Do you want to update to ", latest.Version, "? (y/n): ")
+	return latest, err
+}
+
+func askUpdateForUser(latestVersion *selfupdate.Release) bool {
+	fmt.Print("Do you want to update to ", latestVersion.Version.String(), "? (y/n): ")
 	input, err := bufio.NewReader(os.Stdin).ReadString('\n')
 	if err != nil || (input != "y\n" && input != "n\n") {
-		fmt.Println("Invalid input")
-		return
+		logrus.Error("Invalid input")
+		return false
 	}
 	if input == "n\n" {
-		return
+		logrus.Warn("Aborted")
+		return false
+	}
+	return true
+}
+
+func selfUpdate() error {
+	latestVersion, err := canUpdate()
+	if err != nil {
+		return err
+	}
+
+	logrus.Info("Current version:", currentVersion)
+	if !updateFlag {
+		ok := askUpdateForUser(latestVersion)
+		if !ok {
+			return nil
+		}
 	}
+
 	exe, err := os.Executable()
 	if err != nil {
-		fmt.Println("Could not locate executable path")
-		return
+		return err
 	}
-	if err := selfupdate.UpdateTo(latest.AssetURL, exe); err != nil {
-		fmt.Println("Error occurred while updating binary:", err)
-		return
+	logrus.Info("Updateing ...")
+	if err := selfupdate.UpdateTo(latestVersion.AssetURL, exe); err != nil {
+		return err
 	}
-	fmt.Println("Successfully updated to version", latest.Version)
+	logrus.Info("Successfully updated to version ", latestVersion.Version)
+	return nil
 }
+
 func newUpdateCmd() *cobra.Command {
-	return &cobra.Command{
+	updateCmd := &cobra.Command{
 		Use:   "update",
 		Short: "Update to the latest version",
-		Run: func(cmd *cobra.Command, args []string) {
-			selfUpdate()
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := selfUpdate()
+			if err != nil {
+				return err
+			}
+			return nil
```

Comment:
If you don't want to wrap up the error, I think this should be enough.
```suggestion
			return selfUpdate()
```